File: EstructurasDeDatos/exercise7/Calculator.cs
namespace EstructurasDeDatos.exercise7;

public class Calculator : IGetBestSelling, IGetTotal
{
    public List<Product> GetBestSelling(List<SalesLine> salesLineList)
    {
        var result = new List<Product>();
       /* var maxSales = -1;
    
        foreach (var line in salesLineList)
        {
            if (line.SalesNum > maxSales)
            {
                maxSales = line.SalesNum;
                result.Clear();
                result.Add(line.Product);
            }
            else if (line.SalesNum == maxSales)
            {
                result.Add(line.Product);
            }
        } */
    
        return result;
    }

    public long GetTotal(List<SalesLine> salesLineList)
    {
        return salesLineList.Sum(line => line.LineTotal);
    }
}

File: EstructurasDeDatosPython/Calculator.py
from typing import List

import SalesLine


class Calculator:    
    def calculate_total_sales(self, sales_line_list: List[SalesLine]) -> int: # type: ignore
        return sum(line.get_line_total() for line in sales_line_list)
    
    def get_best_selling(self, sales_line_list: List[SalesLine]) -> List[Product]: # type: ignore
        if not sales_line_list:
            return []
        
        # Encontrar el máximo de ventas
        max_sales = max(line.sales_num for line in sales_line_list)
        
        # Retornar todos los productos con ese máximo
        return [line.product for line in sales_line_list if line.sales_num == max_sales]
